### PR TITLE
.env for root URL behavior+plausible #6193

### LINKED ISSUE #6193: Add control over root URL behavior
### Is your proposal related to a problem?

Currently, the root URL of a deployment (e.g. the value for `NEXT_PUBLIC_WEBAPP_URL`) redirects to `/auth/login`.

For self-hosters, this may not be a desired behavior.

For example, I'm interested in self-hosting a Cal.com deployment for my own, personal schedule. For my usecase, I _definitely_ don't want `NEXT_PUBLIC_WEBAPP_URL` to present a login screen, because that URL is easily guessable by the public (e.g. if I promote `https://schedule.josh.com/josh` people will probably try accessing `https://schedule.josh.com`).

It would be nice to have a first class feature to allow control over this behavior.

### Describe the solution you'd like

Some desired control functionality would be:
1. Allow me to redirect this URL to a specific user's page (e.g. from the example above `https://schedule.josh.com` > `https://schedule.josh.com/josh`) (easy, less preferable, straightforward with current architecture)
2. Allow me to show a specific user's page (e.g. `https://schedule.josh.com` shows the content of `https://schedule.josh.com/josh`) (harder, most preferable, not straightforward with current architecture?)

### Describe alternatives you've considered

Editing source code to do Proposed Solution 1.

### PR DESCRIPTION
**Pull request by hybes**
## What does this PR do?

Adds 2 .env variables for Self Hosted Plausible URL, as well as root url to redirect users if they access the booking url without a session, this could be set to a particular users booking page, or to a landing page.

Fixes #6193 

**Environment**: Staging(main branch) / Production

## Type of change

- [✓ ] New feature (non-breaking change which adds functionality)
- [ ✓] This change requires a documentation update

## How should this be tested?

Configure the two new .env variables:

PLAUSIBLE_URL
ROOT_REDIR_URL

- You can see that traffic now sends to the self hosted Plausible URL, though you must include the full URL address, including the .js file.
- If you access the main WEBAPP_URL now without a session (incognito works), you should see you get redirected to the specified site.

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -147,3 +147,9 @@ SENTRY_IGNORE_API_RESOLUTION_ERROR=
 NEXT_PUBLIC_APP_NAME="Cal.com"
 NEXT_PUBLIC_SUPPORT_MAIL_ADDRESS="help@cal.com"
 NEXT_PUBLIC_COMPANY_NAME="Cal.com, Inc."
+
+# Root URL redirection
+NEXT_PUBLIC_CUSTOM_ROOT_URL="https://cal.com"
```
Comment (PeerRich) [inline .env.example:152]: this already exists

File: .env.example [hunk 2]
```diff
@@ -147,3 +147,9 @@ SENTRY_IGNORE_API_RESOLUTION_ERROR=
...
+
+# Selfhost Plausible URL with js file
+PLAUSIBLE_URL="https://selfhosted-plausible.cal.com/js/plausible.js"
```
Comment (PeerRich) [inline .env.example:155]: this should by default be the correct plausible url

**Comment by Hybes** [comment]
updated to use env variables, not already in use, and to set default plausible URL

File: .env.example [hunk 3]
```diff
@@ -147,3 +147,9 @@ SENTRY_IGNORE_API_RESOLUTION_ERROR=
 NEXT_PUBLIC_APP_NAME="Cal.com"
 NEXT_PUBLIC_SUPPORT_MAIL_ADDRESS="help@cal.com"
 NEXT_PUBLIC_COMPANY_NAME="Cal.com, Inc."
+
+# Root URL redirection
+ROOT_REDIR_URL="https://cal.com"
```
Comment (PeerRich) [inline .env.example:152]: this exists already, no?

**Comment by PeerRich** [comment]
@Hybes we're probably moving all the app settings into a database to not overload our .env

our goal is to have as little .env variables as possible and adding each app-settings into it doesn't seem like a good idea.

will keep this open but we're likely not merging it in this state -- instead we'll implement an "App Manager" where you can change the settings per instance

**Review by PeerRich — CHANGES_REQUESTED** [review]
blocking until we have moved App-specific .env's into a database (App Manager cc @joeauyeung )

**Pull request closed without merging**